### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by danmanj**
This series of commits converts what were static options decided by compiler flags or command line flags into dynamically reconfigurable options contoled by test parameters.

### CODE REVIEW

File: distbench_netutils.cc [hunk 1]
```diff
@@ -107,12 +107,16 @@ absl::StatusOr<DeviceIpAddress> GetBestAddress(std::string_view netdev,
   int score = -1;
   DeviceIpAddress best_match;
   for (const auto& address : all_addresses) {
-    if (address.isLinkLocal() || address.isLoopback()) {
+    if (address.isLinkLocal()) {
```
Comment (kanthicn1) [inline distbench_netutils.cc:110]: Is there a way to specify that the IPv6 preference? Won't this logic return if there are only IPv6 addresses that are supported? Seems like IPv6 will be stuck at a score of -1?

File: distbench_netutils.cc [hunk 2]
```diff
@@ -107,12 +107,16 @@ absl::StatusOr<DeviceIpAddress> GetBestAddress(std::string_view netdev,
...
       continue;
     }
     int cur_score = (address.isIPv4() == prefer_ipv4);
     // High speed networks in cloudlab are on a private IP range.
     cur_score += address.isPrivate();
+    if (!address.isLoopback()) {
```
Comment (kanthicn1) [inline distbench_netutils.cc:116]: What's the difference between linklocal and loopback?

File: distbench_threadpool.cc [hunk 3]
```diff
@@ -42,40 +67,114 @@ DistbenchThreadpool::DistbenchThreadpool(int nb_threads) {
...
 }
 
-void DistbenchThreadpool::AddWork(std::function<void()> function) {
+void SimpleThreadpool::AddWork(std::function<void()> function) {
   absl::MutexLock m(&mutex_);
   work_queue_.push(function);
 }
 
-#else
+// This is the C-Threadpool from an external library.
```
Comment (kanthicn1) [inline distbench_threadpool.cc:82]: For my understanding, does this comment mean you copied it from AbstractThreadPool from the external library? If yes, why couldn't you link to it and why did you have to copy it?

File: distbench_threadpool.cc [hunk 4]
```diff
@@ -42,40 +67,114 @@ DistbenchThreadpool::DistbenchThreadpool(int nb_threads) {
...
+  } else if (name == "cthread") {
+    return std::make_unique<CThreadpool>(size);
+#ifdef WITH_MERCURY
+  } else if (name == "mercury") {
+    return std::make_unique<MercuryThreadpool>(size);
 #endif
+  } else {
+    return {};
+  }
+}
```
Comment (kanthicn1) [inline distbench_threadpool.cc:178]: Also for my understanding, is it possible to add a one line comment at the top of this file that explains when developers should use a simple thread Vs cthread vs mercuryThread?

**Review by kanthicn1 — APPROVED** [review]
I have left a few comments for my understanding. Code changes look good.

File: distbench_netutils.cc (hunk 2 above)
Comment (danmanj) [reply distbench_netutils.cc:116]: link local is an ipv6 address "Scope" it allows machines to talk to each other, but is not routable. It is used by the ipv6 replacement for e.g. dhcp.

File: distbench_threadpool.cc (hunk 4 above)
Comment (danmanj) [reply distbench_threadpool.cc:178]: the selection comes from the user via the test configuration, so this belongs in user documentation.

File: distbench_threadpool.cc (hunk 3 above)
Comment (danmanj) [reply distbench_threadpool.cc:82]: This comment is just moved around. I want to delete this threadpool from the code. it is ironically not thread safe.

**Pull request merged**